src/main/java/com/palominolabs/metrics/newrelic/CustomMetricAttributeFilter.java

```diff
@@ -0,0 +1,466 @@
+package com.palominolabs.metrics.newrelic;
+
+import com.codahale.metrics.*;
+import com.fasterxml.jackson.core.type.TypeReference;
+import com.fasterxml.jackson.databind.ObjectMapper;
+import com.fasterxml.jackson.dataformat.yaml.YAMLFactory;
+import com.google.common.base.Optional;
+import com.google.common.base.Preconditions;
+import com.google.common.base.Supplier;
+import com.google.common.collect.HashBasedTable;
+import com.google.common.collect.Table;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import javax.annotation.Nullable;
+import java.io.IOException;
+import java.net.URI;
+import java.util.Map;
+
+/**
+ * Convenience implementation of {@link MetricAttributeFilter} with grain level customization.
+ * Two constructors are provided:<br>
+ * # first which receives a {@link Table} indexed by metricName and NewRelic metric (@see NewRelicMetric).
+ * For example:
+ * <pre>
+ *     {@code 
+ *      Table<String, NewRelicMetric, Boolean> metricConfig = ImmutableTable.<String, NewRelicMetric, Boolean>builder()
+ *          .put("metricName1", NewRelicMetric.TIMER_MAX, true)
+ *          .put("metricName1", NewRelicMetric.TIMER_MIN, false)
+ *          .put("metricName2", NewRelicMetric.COUNTER_COUNT, true)
+ *          .build();
+ *
+ *      new CustomMetricAttributeFilter(metricConfig, new AllDisabledMetricAttributeFilter());
+ *     }
+ * </pre>
+ * # second which receives a {@link URI} to a <i>yaml</i> file where metrics configuration is defined in the folowwing format:
+ * <pre>
+ *     metricName1:
+ *          TIMER_MAX: true
+ *          TIMER_MIN: false
+ *     metricName2:
+ *          COUNTER_COUNT: true
+ * </pre>
+ *
+ * Both constructors receives a {@link MetricAttributeFilter} which will be used as fallback for all metrics not specified in configuration.
+ * In case <i>fallback</i> is null, {@link AllDisabledMetricAttributeFilter} will be used.
+ */
+public class CustomMetricAttributeFilter implements MetricAttributeFilter {
```

Comment:
I think many people could/will write "Custom" filters, so I don't want to set a precedent of calling anything that is custom "Custom". (It's trivially true, right? Just like we could call them all Class as well.) I'd prefer Table, or some other concrete descriptor of what this does, not an adjective that describes one fairly ancillary aspect of it.